pkg/steps/source.go

```diff
@@ -41,33 +41,61 @@ const (
 	gopath        = "/go"
 	sshPrivateKey = "/sshprivatekey"
 	sshConfig     = "/ssh_config"
+	oauthToken    = "/oauth-token"
+
+	OauthSecretKey = "oauth-token"
 )
 
+type CloneAuthType string
+
+var (
+	CloneAuthTypeSSH   CloneAuthType = "SSH"
+	CloneAuthTypeOAuth CloneAuthType = "OAuth"
+)
+
+type CloneAuthConfig struct {
+	Secret *coreapi.Secret
+	Type   CloneAuthType
+}
+
+func (c *CloneAuthConfig) determineCloneURI(org, repo string) string {
```

Comment:
maybe just `get` instead of `determine`? it's a simple choice, no fancy process :)